components/layout_2020/dom_traversal.rs

```diff
@@ -409,12 +409,20 @@ where
                         vec.push(PseudoElementContentItem::Text(s.to_string()));
                     },
                     ContentItem::Attr(attr) => {
+                        use selectors::Element;
+
                         let element = element
                             .to_threadsafe()
                             .as_element()
                             .expect("Expected an element");
-                        let attr_val = element
-                            .get_attr(&attr.namespace_url, &LocalName::from(&*attr.attribute));
+                        let attr_name = if element.is_html_element_in_html_document() {
+                            &*attr.attribute.to_ascii_lowercase()
+                        } else {
+                            &*attr.attribute
+                        };
```

Comment:
Can you link to the specification and quote the specification text here that explains this code. In particular, why does this only apply to the HTML element?

I think a match statement would work best here as well.

Reply:
Thank you for the review!

Reply:
Thank you for the review!

As for HTML documents, the case sensitivity of the function is clearly stated [here](https://html.spec.whatwg.org/multipage/semantics-other.html#case-sensitivity-of-the-css-'attr()'-function) which also stated how does attribute selectors be matched, which I believe was implemented in [stylo's css attribute selector](https://github.com/servo/stylo/blob/098f1bef3e231bbf86038c0a2b1ddc31031dc422/selectors/matching.rs#L1263). 

I tried to find specifications further, but it seems like there is not unified detailed specifications of all documents for this, except for general consensus that documents based on XML (e.g. SVG, XHTML, and MATHML) is case sensitive for attributes.

Please let me know if there is anything further I can clarify and address.

Reply:
Sorry, I have quoted the specification. Kindly recheck the changes. Thank you!